packages/api/models/granules.js

```diff
@@ -293,72 +294,89 @@ class Granule extends Manager {
    * Build a granule record.
    *
    * @param {Object} params
-   * @param {AWS.S3} params.s3 - an AWS.S3 instance
    * @param {Object} params.granule - A granule object
    * @param {string} params.executionUrl - A Step Function execution URL
-   * @param {Object} params.provider - Provider object
-   * @param {string} params.workflowStatus - Workflow status
-   * @param {string} params.collectionId - Collection ID for the workflow
-   * @param {string} [params.pdrName] - PDR name for the workflow, if any
-   * @param {Object} [params.error] - Workflow error, if any
-   * @param {Object} [params.processingTimeInfo={}]
+   * @param {string} params.collectionId - Cumulus collection id
+   * @param {string} params.provider - Provider id
+   * @param {number} params.timeToArchive - seconds to post to cmr.
+   * @param {number} [params.timeToPreprocess] -  seconds
+   * @param {integer} [params.productVolume] - sum of the files sizes in bytes
+   * @param {number} [params.duration] - seconds
+   * @param {GranuleStatus} params.status - ['running','failed','completed']
+   * @param {number} params.workflowStartTime
+   * @param {Array<ApiFile>} params.files - files associated with the granule.
+   * @param {Object} [params.error] = {} - workflow error that may have occurred.
+   * @param {string} [params.pdrName]
+   * @param {Object} [params.queryFields] - query fields
+   * @param {Object} [params.processingTimeInfo = {}] - from getExecutionProcessingTimeInfo
+   * @param {number} [params.updatedAt  = Date.now()] -
+   * @param {Object} [params.cmrTemporalInfo = {}] - from cmr.getGranuleTemporalInfo
    *   Info describing the processing time for the granule
-   * @returns {Promise<Object>} A granule record
+   * @returns {Promise<Object>} A dynamoDb granule record
    */
   async generateGranuleRecord({
-    s3,
     granule,
     executionUrl,
     collectionId,
     provider,
+    timeToArchive,
+    timeToPreprocess,
+    productVolume,
+    duration,
+    status,
     workflowStartTime,
+    files = [],
     error,
     pdrName,
-    workflowStatus,
     queryFields,
     processingTimeInfo = {},
     updatedAt,
+    cmrTemporalInfo = {},
   }) {
-    if (!granule.granuleId) throw new CumulusModelError(`Could not create granule record, invalid granuleId: ${granule.granuleId}`);
+    if (!granule.granuleId) {
+      throw new CumulusModelError(`Could not create granule record, invalid granuleId: ${granule.granuleId}`);
+    }
 
     if (!collectionId) {
       throw new CumulusModelError('collection required to generate a granule record');
     }
 
+    if (!files || files.length === 0) {
```

Comment:
Interesting, I assumed it was a requirement from what I was seeing.  Long story, not short. 


on origin/master: If files is undefined it creates the postgres record correctly because it calls.
`_generateFilesFromGranule` which provides an empty files array as a default.
https://github.com/nasa/cumulus/blob/master/packages/api/lambdas/sf-event-sqs-to-db-records/write-granules.js#L370

But inside _writeGranules when it stores the dynamoDB record it calls `storeGranuleFromCumulusMessage` and in turn `generateGranuleRecord` calls `buildDatabaseFiles` directly with an undefined files.
https://github.com/nasa/cumulus/blob/master/packages/api/lib/FileUtils.js#L129

in the test it's mocked to just resolve as the input files. 
    buildDatabaseFiles: (params) => Promise.resolve(params.files),

But in reality a call to buildDatabaseFiles with an undefined files will throw an `UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'map' of undefined`

I *think* that will propagate up making the trx transaction fail, and keeping either of the granule records from being written.

So, I probably guessed wrong when I assumed this was the expected behavior? I don't know if it makes sense to have a granule without a file, but can make this work either way.

Reply:
Hmm, I'm wondering if that's just a bug if we're calling it that way.  The most salient user of that method is this

```
const translateGranule = async (
  granule,
  fileUtils = FileUtils
) => {
  if (isNil(granule.files)) return granule;

  return {
    ...granule,
    files: await fileUtils.buildDatabaseFiles({
      s3: awsClients.s3(),
      files: granule.files,
    }),
  };
};```

which has a catch re: taht value being nil.

Reply:
fef362ebef